partners_api/ads_engine.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "partners_api/ads_base.hpp"
+#include "partners_api/banner.hpp"
+
+#include "base/singleton.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace ads
+{
+class Collector;
+using Engine = base::Singleton<Collector>;
+
+class Collector
+{
+public:
+  Collector();
+
+  bool HasBanner(feature::TypesHolder const & types) const;
+  std::vector<Banner> GetBanners(feature::TypesHolder const & types) const;
+
+private:
+  using ContainerPtr = std::unique_ptr<IContainer>;
```

Comment:
Do you really need this type alias? It's used only once and in the private section.